planner/core/planbuilder.go

```diff
@@ -1790,7 +1790,10 @@ func (b *PlanBuilder) buildAdminCheckTable(ctx context.Context, as *ast.AdminStm
 			}
 		}
 		if idx == nil {
-			return nil, errors.Errorf("index %s do not exist", as.Index)
+			if idxName == "primary" && tbl.Meta().PKIsHandle {
+				return nil, errors.Errorf("index `primary` is not defined when primary key is a single integer column and clustered")
+			}
+			return nil, errors.Errorf("index %s does not exist", as.Index)
```

Comment:
```suggestion
			if idxName == "primary" && tbl.Meta().PKIsHandle {
				return nil, errors.Errorf("index `primary` is not defined when primary key is a single integer column and clustered")
			}
			return nil, errors.Errorf("index %s does not exist", as.Index)
```

How about changing the error message to "secondary index %s does not exist"? I think it is much clear than "index `primary` is not defined when primary key is a single integer column and clustered".

Reply:
Do you mean ADMIN CHECK/RECOVER/CLEANUP INDEX is only used for secondary index ?

Reply:
@taofengliu `ADMIN * INDEX` can be also used for non-clustered primary key(I usually consider it as a special secondary index).